### PR TITLE
Fix Java 11 build

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
- Fix failed test with Java 11 
See #1038 

Root cause is an known issue in wiremock with Java 11 https://github.com/tomakehurst/wiremock/issues/1009. Changed `CONNECTION_RESET_BY_PEER` to `RANDOM_DATA_THEN_CLOSE` instead to fix the test.

- Fix false positive bugs reported by spotbugs

- Update travis script to run java 11 for obvious reasons.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: http-clients/netty-nio-client/src/test/resources/jetty-logging.properties [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+# Set up logging implementation
+org.eclipse.jetty.util.log.class=org.eclipse.jetty.util.log.StdErrLog
+org.eclipse.jetty.LEVEL=WARN
```
Comment (dagnir) [inline http-clients/netty-nio-client/src/test/resources/jetty-logging.properties]: minor: How much extra logging will this create? Just worried about the log size in Travis

**Comment by zoewangg** [comment]
The test is happy but spotbugs failed :(

File: http-clients/netty-nio-client/src/test/resources/jetty-logging.properties (hunk 1 above)
Comment (zoewangg) [reply http-clients/netty-nio-client/src/test/resources/jetty-logging.properties]: Good point, will change it to `OFF`

File: aws-sdk-java/pom.xml [hunk 2]
```diff
@@ -15,6 +15,19 @@
         Amazon AutoScaling, etc).
     </description>
     <url>https://aws.amazon.com/sdkforjava</url>
+
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>software.amazon.awssdk</groupId>
+                <artifactId>bom-internal</artifactId>
```
Comment (dagnir) [inline aws-sdk-java/pom.xml]: Why does this need the BOM?

Comment (zoewangg) [reply aws-sdk-java/pom.xml]: This is something I tried to make it build with CodeBuild on this module. After this change, this module can be built successfully but other module fails. I can remove this.

Comment (dagnir) [reply aws-sdk-java/pom.xml]: Ah okay. Yeah let's remove it for now, this module shouldn't need it.

**Pull request merged**